src/User.cpp

```diff
@@ -68,13 +70,24 @@ bool operator==(const User &lhs, const User &rhs)
             lhs.firecode_ == rhs.firecode_);
 }
 
-std::string User::getPassword() const
+std::string User::getEmail() const
 {
-    return password_;
+    return email_;
 }
 
+bool User::validatePassword(const std::string& password) const {
+    return !hashedPassword_.compare(hash_.generateHash(salt_, password));
+}
 
-std::string User::getEmail() const
-{
-    return email_;
+bool User::validateEmail(const std::string& email) const {
+    if (!email.empty() && (email.size() == email_.size())) {
+        for(std::size_t i=0; i<email.size(); ++i) {
+            if(tolower(email[i]) != tolower(email_[i]))
+                return false;            
+        }
+    }
+    else {
+        return false;
+    }
+    return true;
```

Comment:
is this function unit tested? ;>

Reply:
At the moment it is in-directly tested through the AuthenticationProviderTests (validateEmail and validatePassword member functions are used by the () operator of AuthenticationProvider class) :)